new module: AIX rootvg backup image using mksysb

kairoaraujo opened the pull request:
##### SUMMARY
This module is simple but very useful for AIX system
administrators. Easy to construct playbooks to generate
and manage rootvg backups using mksysb tool.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
mksysb
##### ANSIBLE VERSION
```
ansible 2.5.0 (mksysb d2d7d3205b) last updated 2017/09/16 21:28:15 (GMT +200)
  config file = None
  configured module search path = ['/Users/kairoaraujo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kairoaraujo/Dev/mksysb/ansible/lib/ansible
  executable location = /Users/kairoaraujo/Dev/mksysb/ansible/bin/ansible
  python version = 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06) [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```


##### ADDITIONAL INFORMATION
None

kairoaraujo committed:
new module: AIX rootvg backup image using mksysb

This module is simple but very useful for AIX system
administrators. Easy to construct playbooks to generate
and manage rootvg backups using mksysb tool.

jtyr commented:
Dude, first finish the first module (#30381) and then open another PR. I'm not gonna give you the same comments on both PRs. I'm expecting you to learn from the first PR and incorporate all changes here as well.

kairoaraujo commented:
@jtyr I'm very glad about comments that you're doing. I will apply here as well.
Sorry but, the examples that I got weren't too strict as those that you're doing. 

Anyway, to have a good quality code, I agree that it needs to be made and I will do my best.

Also, the docs are not clear about it: http://docs.ansible.com/ansible/latest/community.html 

>Code developed for Ansible needs to support Python2-2.6 or higher and Python3-3.5 or higher.
>Use a 4-space indent, not tabs.
>We do not enforce 80 column lines; up to 160 columns are acceptable.
>We do not accept ‘style only’ pull requests unless the code is nearly unreadable.
>We are “PEP8ish”, but not strictly compliant, see testing_pep8 for more information.

kairoaraujo commented on lib/ansible/modules/system/mksysb.py:
```diff
@@ -0,0 +1,198 @@
+        module.exit_json(changed=True, msg=mksysb_output)
+    else:
+        module.fail_json(msg="mksysb failed.", rc=rc, err=err)
+
```
I'm planning to use supports_check_mode here.
Changing this block to:

```python
    # Validate if storage_path is a valid directory.
    if os.path.isdir(storage_path):
        if not module.check_mode:
            # Generates the mksysb image backup.
            mksysb_cmd = module.get_bin_path("mksysb", True)
            rc, mksysb_output, err = module.run_command(
                "%s -X %s %s %s %s %s %s %s %s %s %s/%s" % (
                    mksysb_cmd, create_map_files, use_snapshot, exclude_files,
                    exclude_wpar_files, software_packing, extended_attrs,
                    backup_crypt_files, backup_dmapi_fs, new_image_data,
                    storage_path, name))
            if rc == 0:
                module.exit_json(changed=True, msg=mksysb_output)
            else:
                module.fail_json(msg="mksysb failed.", rc=rc, err=err)

        module.exit_json(changed=True)

    else:
        module.fail_json(msg="Storage path %s is not valid." % storage_path)
```

When I give ok for this block I will commit the changes to my fork/branch

kairoaraujo reviewed (commented):
Just some changes I already did

kairoaraujo committed:
added module_check, pep8, non-written convention

- implemented module_check;
- fixed some pep8 and non-written convention

jtyr commented on lib/ansible/modules/system/mksysb.py:
```diff
@@ -0,0 +1,208 @@
+            software_packing=dict(type='bool', default=False),
+            storage_path=dict(required=True),
+            use_snapshot=dict(type='bool', default=False)
+        ), supports_check_mode=True,
```
I would put the `supports_check_mode=True,` on the next line.

jtyr commented on lib/ansible/modules/system/mksysb.py:
```diff
@@ -0,0 +1,208 @@
+import os
+
+# Command options.
+map_file_opt = {
```
I would put all these variables inside the `main()` function as there is no reason to have them global because you don't share them across multiple functions.

jtyr commented on lib/ansible/modules/system/mksysb.py:
```diff
@@ -0,0 +1,208 @@
+    default: "no"
+  exclude_files:
+    description:
+      - Excludes files using /etc/rootvg.exclude.
```
The file path [should be](http://ansible-docs.readthedocs.io/zh/stable-2.0/rst/developing_modules.html#module-checklist) decorated by `C()`. The same for `software_packing`.

jtyr reviewed (changes requested):
Otherwise it looks good.

kairoaraujo committed:
removed parameters as global variables and doc

Moved global variables parameters to inside main()
Better doc format for mentioned files

jtyr reviewed (approved):
LGTM. `shipit`

adejoux commented:
shipit
